daisy_workflows/build-publish/rhui/rhua_sync_status.py

```diff
@@ -0,0 +1,154 @@
+#  Copyright 2022 Google Inc. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Publish metrics for RHUA sync status."""
+
+import os
+import subprocess
+import time
+
+import requests
+
+
+REPODIR = '/var/lib/rhui/remote_share/symlinks/pulp/content/content'
+
+
+def GetRepodataList(dirname):
+  # Use "find" here because os.walk takes ~30 minutes,
```

Comment:
if this works i'm fine with it, but i think we would not simply say os.walk takes too long but *why* it takes too long. no reason python can't be as fast as find

Reply:
So, I'm not really sure why os.walk takes as long as it does. The times listed came from actually trying to use it. I tested both find and os.walk to simply generate a list of "repodata" directories, and while the call to find usually takes less than 20 seconds, using os.walk took about half an hour. It's generally expected that os.walk would be slower than find, since find apparently has some optimizations, but it's possible that there is something specific about this directory that makes it worse. (Apparently pre-Python 3.5, os.walk was particularly bad because it would stat each file. That was fixed in 3.5 with os.scandir, but apparently os.walk is still much slower than find in some cases.)

Reply:
Removed the comment.